cpp/onedal/detail/common.hpp

```diff
@@ -33,5 +33,22 @@ using shared = std::shared_ptr<T>;
 template <typename T>
 using pimpl = shared<T>;
 
+struct pimpl_accessor {
+  template <typename Object>
+  auto& operator()(Object&& object) const {
+    return object._impl;
+  }
+};
+
+template <typename Impl, typename Object>
+Impl& get_impl(Object&& object) {
```

Comment:
Couldn't we use `auto` for return type? Explicit `Impl` forces a caller specify impl type. It's redundant, because impl type can be inferred from the `Object` (assuming object is pimpl).

Reply:
This method designed to return the type specified by user. To get a auto-inferred type, use `get_impl_ptr()`